### ISSUE TITLE
Issue on page /rst/developer_guide/dev_install.html 

### ISSUE THREAD

**Issue by gritk** [eradiate/eradiate#456]
Please could you also specify the version of Ubuntu etc. that you are using.

**The following additional steps are required for Ubuntu 22.04:**  

**Citation of the https://stackoverflow.com/questions/66223241/how-to-install-clang-11-on-debian**

Add the following lines to your /etc/apt/sources.list:

          deb http://apt.llvm.org/buster/ llvm-toolchain-buster main 
          deb-src http://apt.llvm.org/buster/ llvm-toolchain-buster main 
          deb http://apt.llvm.org/buster/ llvm-toolchain-buster-10 main 
          deb-src http://apt.llvm.org/buster/ llvm-toolchain-buster-10 main 
          deb http://apt.llvm.org/buster/ llvm-toolchain-buster-11 main 
          deb-src http://apt.llvm.org/buster/ llvm-toolchain-buster-11 main

Add signatures for these repos (otherwise apt-get update will complain in the next step)

          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -

Run apt-get update to add these new repos to the apt-get:

         sudo apt-get update

Install clang-11:

          sudo apt-get install clang-11

**https://stackoverflow.com/questions/54587052/cmake-on-mac-could-not-find-threads-missing-threads-found
            # assume built-in pthreads on MacOS (Works for me in Ubuntu as well)**

            IF(APPLE )
                set(CMAKE_THREAD_LIBS_INIT "-lpthread")
                set(CMAKE_HAVE_THREADS_LIBRARY 1)
                set(CMAKE_USE_WIN32_THREADS_INIT 0)
                set(CMAKE_USE_PTHREADS_INIT 1)
                set(THREADS_PREFER_PTHREAD_FLAG ON)
            ENDIF()

**see - https://github.com/ClickHouse/ClickHouse/issues/7729 - The install of the mitsuba target requires changing an RPATH from the build   tree, but this is not supported with the Ninja generator unless on an   ELF-based or XCOFF-based platform.  The CMAKE_BUILD_WITH_INSTALL_RPATH   variable may be set to avoid this relinking step.**

           Resolved after removing CMakeCache.txt file

**Comment by leroyvn** [comment]
Hi @gritk, thanks a lot for raising and investigating these issues.

## Our Linux dev setup

We internally run Ubuntu 22.04 (jammy) and 24.04 (noble):

* jammy has a [clang-11](https://packages.ubuntu.com/jammy/clang-11) in the `universe` component, which is activated on default installations to my knowledge.
* noble doesn't have it, but provides [clang-14](https://packages.ubuntu.com/noble/clang-14) to [clang-18](https://packages.ubuntu.com/noble/clang-18).

Indeed sourcing packages from LLVM's repos is the thing to do to have access to all versions, including "old" ones. 

Despite what is said on our dev build page, all "reasonably recent" clang versions seem to work (I personally use clang 11 and 15 depending on the machine). We will update the documentation to make compiler requirements less stringent as soon as the next major Mitsuba update is released (hopefully next month).

## Our macOS dev setup

We run macOS 15 (sequoia) and build with XCode's clang (based on clang 19 as of October 2024 IIRC). I have never experienced pthreads or rpath issues so far, thank you for troubleshooting this.

**Comment by leroyvn** [comment]
Indeed, this is something we'll change: we can be much less strict about it. Sorry for the trouble!

**Comment by manikbali** [comment]
Thanks: I am trying to run the rpv BSDF. Can you share a code instance of using this function. Also eradiate data fetch minimal
Usage: eradiate data [OPTIONS] COMMAND [ARGS]...
Try 'eradiate data --help' for help.
╭─ Error ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ No such command 'fetch'.                                                                                                                                                                  │
╰────────────────────────────────────────────

**Comment by leroyvn** [comment]
Hi @manikbali, you seem to have several issues you need help with. To keep this issue appropriately scoped (I am actually about to close it because I addressed @gritk's point), could you please open one or several [discussions](https://github.com/eradiate/eradiate/discussions)? This is the best place to report usage issues. Thank you for your understanding.

**Comment by leroyvn** [comment]
Hi @gritk, I updated the installation instructions. The latest documentation version should now be less strict about the required Clang version. I'll close this, feel free to reopen if needed.
